pkg/monitoring/metrics/operator_metrics.go

```diff
@@ -0,0 +1,143 @@
+package metrics
+
+import (
+	"strings"
+
+	"github.com/machadovilaca/operator-observability/pkg/operatormetrics"
+	ioprometheusclient "github.com/prometheus/client_model/go"
+)
+
+const (
+	counterLabelCompName = "component_name"
+	counterLabelAnnName  = "annotation_name"
+
+	hyperConvergedExists    = 1.0
+	hyperConvergedNotExists = 0.0
+)
+
+const (
+	SystemHealthStatusHealthy float64 = iota
+	SystemHealthStatusWarning
+	SystemHealthStatusError
+)
+
+var (
+	operatorMetrics = []operatormetrics.Metric{
+		overwrittenModifications,
+		unsafeModifications,
+		hyperConvergedCRExists,
+		systemHealthStatus,
+	}
+
+	overwrittenModifications = operatormetrics.NewCounterVec(
+		operatormetrics.MetricOpts{
+			Name: "kubevirt_hco_out_of_band_modifications_total",
+			Help: "Count of out-of-band modifications overwritten by HCO",
+		},
+		[]string{counterLabelCompName},
+	)
+
+	unsafeModifications = operatormetrics.NewGaugeVec(
+		operatormetrics.MetricOpts{
+			Name: "kubevirt_hco_unsafe_modifications",
+			Help: "Count of unsafe modifications in the HyperConverged annotations",
+		},
+		[]string{counterLabelAnnName},
+	)
+
+	hyperConvergedCRExists = operatormetrics.NewGauge(
+		operatormetrics.MetricOpts{
+			Name: "kubevirt_hco_hyperconverged_cr_exists",
+			Help: "Indicates whether the HyperConverged custom resource exists (1) or not (0)",
+		},
+	)
+
+	systemHealthStatus = operatormetrics.NewGauge(
+		operatormetrics.MetricOpts{
+			Name: "kubevirt_hco_system_health_status",
+			Help: "Indicates whether the system health status is healthy (0), warning (1), or error (2), by aggregating the conditions of HCO and its secondary resources",
+		},
+	)
+)
+
+// IncOverwrittenModifications increments counter by 1
+func IncOverwrittenModifications(kind, name string) {
+	overwrittenModifications.WithLabelValues(getLabelsForObj(kind, name)).Inc()
+}
+
+// GetOverwrittenModificationsCount returns current value of counter. If error is not nil then value is undefined
```

Comment:
> If error is not nil then value is undefined

It looks like the value returned is 0 if the error is not nil. Is 0 analogous to "undefined" in this context?

Reply:
not sure what you asking here.. if the error is not nil then it's not defined. 0 is the default value if error is nil.

Reply:
This pr refactor the monitoring code but not supposed to change it. so this is how it was before this pr as well.
but it does make sense, maybe we should only update the comment to `If error is not nil then value is 0`
@machadovilaca WDYT?

Reply:
> maybe we should only update the comment

Yes, that's what I was pointing to. Only the comment part.

Reply:
i think we can consider the 0 to be an `undefined` value, since it has no meaning here
previous `GetMetricValue` was already returning 0 in case of error

Reply:
+1 so let's leave as is. closing this issue